tests/functional/build/test_build.py

```diff
@@ -0,0 +1,296 @@
+        "snap_0.sql": snapshots__snap_0_sql,
+        "snap_99.sql": snapshots__snap_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def tests_failing():
+    return {
+        "model_2.sql": tests_failing__model_2_sql,
+        "test.yml": tests_failing__test_yml,
+        "model_0.sql": tests_failing__model_0_sql,
+        "model_1.sql": tests_failing__model_1_sql,
+        "model_99.sql": tests_failing__model_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def models():
+    return {
+        "model_2.sql": models__model_2_sql,
+        "test.yml": models__test_yml,
+        "model_0.sql": models__model_0_sql,
+        "model_1.sql": models__model_1_sql,
+        "model_99.sql": models__model_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def models_failing():
+    return {
+        "model_3.sql": models_failing__model_3_sql,
+        "model_2.sql": models_failing__model_2_sql,
+        "test.yml": models_failing__test_yml,
+        "model_0.sql": models_failing__model_0_sql,
+        "model_1.sql": models_failing__model_1_sql,
+        "model_99.sql": models_failing__model_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def seeds():
+    return {"countries.csv": seeds__countries_csv}
+
+
+@pytest.fixture(scope="class")
+def models_circular_relationship():
+    return {
+        "test.yml": models_circular_relationship__test_yml,
+        "model_0.sql": models_circular_relationship__model_0_sql,
+        "model_1.sql": models_circular_relationship__model_1_sql,
+        "model_99.sql": models_circular_relationship__model_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def models_simple_blocking():
+    return {
+        "schema.yml": models_simple_blocking__schema_yml,
+        "model_b.sql": models_simple_blocking__model_b_sql,
+        "model_a.sql": models_simple_blocking__model_a_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def models_interdependent():
+    return {
+        "schema.yml": models_interdependent__schema_yml,
+        "model_c.sql": models_interdependent__model_c_sql,
+        "model_a.sql": models_interdependent__model_a_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def project_files(
+    project_root,
+    snapshots,
+    tests_failing,
+    models,
+    models_failing,
+    seeds,
+    models_circular_relationship,
+    models_simple_blocking,
+    models_interdependent,
+):
+    write_project_files(project_root, "snapshots", snapshots)
+    write_project_files(project_root, "tests-failing", tests_failing)
+    write_project_files(project_root, "models", models)
+    write_project_files(project_root, "models-failing", models_failing)
+    write_project_files(project_root, "seeds", seeds)
+    write_project_files(project_root, "models-circular-relationship", models_circular_relationship)
+    write_project_files(project_root, "models-simple-blocking", models_simple_blocking)
+    write_project_files(project_root, "models-interdependent", models_interdependent)
```

Comment:
None of this is needed.  Just define the models/snapshots/etc in the class.  The project fixture takes care of the `write_project_files()` function.  I'll add an example below of what I mean.

Reply:
the reason we were having different folders for different model was that there could be only one set of files for the whole tests so we put everything in different folders and use `model_path` to point to it. In the new test framework, we write a project repo to disk for every test class, so you can define the `models` fixture that only containes files needed for current class, and not need the `model_path` any more.
https://github.com/dbt-labs/dbt-core/blob/ab0c3510ebe70a67fb669b88e6ea36aff2f46339/tests/functional/schema_tests/test_schema_v2_tests.py#L41-L43